File: 1009 - Complement of Base 10 Integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        b = bin(n)
        b = b[2:]
        b = [c for c in b]
        for i in range(len(b)):
            if b[i] == '0':
                b[i] = '1'
            else:
                b[i] = '0'
        b = '0b' + ''.join(b)
        res = int(b, base=0)
        return res

File: 1009 - Complement of Base 10 Integer.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int bitwiseComplement(int n) {
        List<Integer> bin = new ArrayList<>();
        int res = 0;
        if (n == 0) return 1;
        while (n > 0) {
            bin.add(n % 2);
            n /= 2;
        }
        for (int i = 0; i < bin.size(); i++) {
            bin.set(i, bin.get(i) ^ 1);
            res += bin.get(i) * Math.pow(2, i);
        }
        return res;
    }
}